examples/cis3-nft-sponsored-txs/src/lib.rs

```diff
@@ -609,18 +613,35 @@ fn contract_view<S: HasStateApi>(
             operators,
         }));
     }
+
     let all_tokens = state.all_tokens.iter().map(|x| *x).collect();
 
+    let mut all_nonces = Vec::new();
+    for (account_address, nonce) in state.nonces.iter() {
+        all_nonces.push((*account_address, *nonce));
+    }
+
+    let mut all_public_keys = Vec::new();
+    for (account_address, public_key) in state.public_key_registry.iter() {
+        all_public_keys.push((*account_address, *public_key));
+    }
+
     Ok(ViewState {
         state: inner_state,
         all_tokens,
+        all_nonces,
+        all_public_keys,
     })
 }
 
-/// Register a public key to a given account. The corresponding private
+/// The parameter type for the contract function `registerPublicKey`.
+#[derive(Debug, Serialize, SchemaType)]
+pub struct RegisterPublicKeyParams(#[concordium(size_length = 2)] pub Vec<RegisterPublicKeyParam>);
```

Comment:
If you use a set (fx `BTreeSet`), then you can avoid duplicates automatically.
We also support sets in the schema, and our tools (at least concordium-client) will check whether it contains any duplicates before making a transaction.

Reply:
I kept it for now since it is the same pattern as the `balanceOf` and `operatorOf` functions to keep it consistent.